### DISCUSSION TITLE
what's the difference btween this project and openai code interpreter?

### DISCUSSION THREAD

**Discussion by Bellafc** [ricklamers/gpt-code-ui#44] [Q&A]
It seems when I input the same questions in this platform and the gpt4 code interpreter, the answers are far different. Can I understand that as this project adds some prompts based on gpt 4 or 3.5, but the code interpreter has some more built-in that only Openai knows?

**Comment by ricklamers** [comment]
You're right. Code Interpreter internals aren't fully disclosed (e.g. which prompts they're using). They likely also have additional logic implemented around code execution and execution parsing (and feeding that into the context window for consecutive generation).

Furthermore, they likely also use a different model as hinted at by the `?model=gpt-4-code-interpreter` parameter.

**Reply by Bellafc** [reply]
Thank you so so much for the prompt reply! But may I ask is the `mode=gpt-4-code-interpreter`  available now? When I tried to use this model, it raised an error `InvalidRequestError:The model gpt-4-code-interpreter does not exist`

**Reply by ricklamers** [reply]
Nope, don't think it has an API
